03week/ticTacToe.js

```diff
@@ -3,55 +3,155 @@
 const readline = require('readline');
 const rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
+    input: process.stdin,
+    output: process.stdout
 });
 let board = [
-  [' ', ' ', ' '],
-  [' ', ' ', ' '],
-  [' ', ' ', ' ']
+    [' ', ' ', ' '],
+    [' ', ' ', ' '],
+    [' ', ' ', ' ']
 ];
 
 let playerTurn = 'X';
 
 function printBoard() {
-  console.log('   0  1  2');
-  console.log('0 ' + board[0].join(' | '));
-  console.log('  ---------');
-  console.log('1 ' + board[1].join(' | '));
-  console.log('  ---------');
-  console.log('2 ' + board[2].join(' | '));
+    console.log('   0  1  2');
+    console.log('0 ' + board[0].join(' | '));
+    console.log('  ---------');
+    console.log('1 ' + board[1].join(' | '));
+    console.log('  ---------');
+    console.log('2 ' + board[2].join(' | '));
 }
 
+//programming notes
+// player x enters a row and column
+
+// check for valid entries
+// is space already filled?(technically we could start checking this only after x has taken its first turn and o makes first selection)
+// update board with x
+// switch players
+// player o enters a row and column
+// check for valid entries
+// is space already filled?
+// update board with o
+// switch players
+//
+// after player x has gone 3 times, check for win... and from now on check for win after every turn
+//check for a win - needs to look at xxx or ooo in each of 3 horizontal rows, each of 3 vertical columns, and each of 2 diagonals.
+
+
+// delcare player x or o a winner
+// if niether player wins when all squares are filled, declare "cats game"
+
+//Had a thought last night about using the modulo check if even in order to determine wether it was x or o's turn... may not end up needing this, but a thought.
+
+
+
+
+
+
 function horizontalWin() {
-  // Your code here
+    // Your code here
+    for (let i = 0; i < 3; i++) {
+        if (board[i][0] == board[i][1] && board[i][1] == board[i][2] && board[i][0] == board[i][2]) {
+            return true;
+        }
+    }
+    return false;
 }
 
 function verticalWin() {
-  // Your code here
+    // Your code here
+    for (let i = 0; i < 3; i++) {
+        if (board[0][i] == board[1][i] && board[1][i] == board[2][i] && board[0][i] == board[2][i]) {
+            return true;
+        }
+    }
+    return false;
 }
 
 function diagonalWin() {
-  // Your code here
+    // Your code here
+    if (board[0][0] == board[1][1] && board[1][1] == board[2][2] && board[0][0] == board[2][2]) {
+        return true;
+    }
+    if (board[0][2] == board[1][1] && board[1][1] == board[2][0] && board[0][2] == board[2][0]) {
+        return true;
+    }
+
 }
 
 function checkForWin() {
-  // Your code here
+    // Your code here
+    const blankInputValue = ' ';
+
+
+    if (horizontalWin() || verticalWin() || diagonalWin()) {
+        console.log("Player " + playerTurn + " wins!");
+            //maybe return True here, then have another function declare a winner or tie.
```

Comment:
Great coding very advanced way of using splice to store your users inputs. only code missing is checking for a tie.